### PR TITLE
Add docs for DRAW_ORDER_REVERSE_LIFETIME constant and minor XR log improvement

### PR DESCRIPTION
**Pull request by Anutrix**
Added docs for DRAW_ORDER_REVERSE_LIFETIME constant and minor XR log improvement.

The changes in `xr_server.cpp` is made to match `text_server.cpp` which improves readability of the interface name.
```
XR: Removed interfaceNative mobile
XR: Removed interfaceOpenXR
```
is changed to:
```
XR: Removed interface "Native mobile"
XR: Removed interface "OpenXR"
```

### CODE REVIEW

File: doc/classes/GPUParticles2D.xml [hunk 1]
```diff
@@ -132,6 +132,7 @@
 			Particles are drawn in order of remaining lifetime.
 		</constant>
 		<constant name="DRAW_ORDER_REVERSE_LIFETIME" value="2" enum="DrawOrder">
+			Particles are drawn in reverse order of remaining lifetime.
```
Comment (Calinou) [inline doc/classes/GPUParticles2D.xml:135]: I would add a concrete example for this (also for Lifetime). Does this mean the newest particles are drawn in front, or the oldest particles?

File: doc/classes/RenderingServer.xml [hunk 2]
```diff
@@ -4450,6 +4450,7 @@
 			Sort particles based on their lifetime.
 		</constant>
 		<constant name="PARTICLES_DRAW_ORDER_REVERSE_LIFETIME" value="2" enum="ParticlesDrawOrder">
+			Sort particles based on the inverse of their lifetime.
```
Comment (Calinou) [inline doc/classes/RenderingServer.xml:4453]: Same as above.

File: doc/classes/GPUParticles2D.xml (hunk 1 above)
Comment (Anutrix) [reply doc/classes/GPUParticles2D.xml]: Made the changes. Not sure if I worded it right. Do check if they are fine.

File: doc/classes/RenderingServer.xml (hunk 2 above)
Comment (YuriSizov) [reply doc/classes/RenderingServer.xml:4453]: @Anutrix This hasn't been updated to match the others.

File: doc/classes/CPUParticles2D.xml [hunk 3]
```diff
@@ -297,7 +297,8 @@
 			Particles are drawn in the order emitted.
 		</constant>
 		<constant name="DRAW_ORDER_LIFETIME" value="1" enum="DrawOrder">
-			Particles are drawn in order of remaining lifetime.
+			Particles are drawn in order of remaining lifetime. In other words, the particle with the most lifetime is drawn on the front.
```
Comment (AThousandShips) [inline doc/classes/CPUParticles2D.xml:300]:
```suggestion
			Particles are drawn in order of remaining lifetime. In other words, the particle with the highest lifetime is drawn at the front.
```
General grammar, "most" is used for countable things, like "most points", this is uncountable, can also be "greatest"

Same everywhere

File: doc/classes/CPUParticles2D.xml [hunk 4]
```diff
@@ -297,7 +297,8 @@
...
+			E.g, if each particle has same lifetime, newest particle is drawn on the front as it has the most lifetime at any given point.
```
Comment (AThousandShips) [inline doc/classes/CPUParticles2D.xml:301]:
```suggestion
			E.g, if each particle has the same lifetime, the newest particle is drawn at the front as it has the highest lifetime at any given point.
```
This is largely just the same as above reworded, don't think both are needed

File: doc/classes/GPUParticles2D.xml [hunk 5]
```diff
@@ -129,9 +129,12 @@
 			Particles are drawn in the order emitted.
 		</constant>
 		<constant name="DRAW_ORDER_LIFETIME" value="1" enum="DrawOrder">
-			Particles are drawn in order of remaining lifetime.
+			Particles are drawn in order of remaining lifetime. In other words, the particle with the most lifetime is drawn on the front.
+			E.g, if each particle has same lifetime, newest particle is drawn on the front as it has the most lifetime at any given point.
 		</constant>
 		<constant name="DRAW_ORDER_REVERSE_LIFETIME" value="2" enum="DrawOrder">
+			Particles are drawn in reverse order of remaining lifetime. In other words, the particle with the least lifetime is drawn on the front.
```
Comment (AThousandShips) [inline doc/classes/GPUParticles2D.xml:136]:
```suggestion
			Particles are drawn in reverse order of remaining lifetime. In other words, the particle with the lowest lifetime is drawn at the front.
```
Same for all other ones of this too

Alternatively "smallest"

File: doc/classes/GPUParticles2D.xml [hunk 6]
```diff
@@ -129,9 +129,12 @@
...
+			E.g, if each particle has same lifetime, oldest particle is drawn on the front as it has the least lifetime at any given point.
```
Comment (AThousandShips) [inline doc/classes/GPUParticles2D.xml:137]:
```suggestion
			E.g, if each particle has the same lifetime, the oldest particle is drawn at the front as it has the lowest lifetime at any given point.
```
Same note about redundancy as above

**Comment by AThousandShips** [comment]
Please rebase your branch, there might be some issue with out of date references with the codespell package

**Comment by Anutrix** [comment]
> Please rebase your branch, there might be some issue with out of date references with the codespell package (over a month out of date branch)

Done.

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.1.4.

**Pull request merged**